src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: Array<T>, key: keyof T): GroupsMap<T> {
+  const grouppedData: GroupsMap<T> = {};
+
+  items.forEach((item) => {
```

Comment:
i think will be better use `reduce` for this case without `grouppedData` variable